### ISSUE TITLE
lint() should lint all files 

### ISSUE THREAD

**Issue by hadley** [jimhester/lintr#217]
Below the current directory with `R|r|Rmd|rmd` extension.

**Comment by hadley** [comment]
Or possibly renamed `lint_package()` to link project.

**Comment by naupaka** [comment]
+1 this would be great

**Comment by mattiaciollaro** [comment]
This would really be a nice feature to add.
I understand currently there are only two options:

- lint a single file

- lint an R package.

It would be nice to have a method that simply lints all `.r` and `.R` files in a given subtree.
(I believe the majority of linters for other programming languages allow to do this?)

In my case, the lack of this functionality is making it very difficult to integrate `lintr` with, e.g.,  CircleCI :(

**Comment by naupaka** [comment]
I have a little R script that I use with Travis to lint all the Rmd files:

```r
# find all .Rmd files
f <- list.files(recursive = TRUE)
rmds <- f[grepl(".Rmd$", f)]

# output lints
for (rmdfile in rmds) {
  print(lintr::lint(rmdfile))
}

# expect no lints
for (rmdfile in rmds) {
  lintr::expect_lint(checks = NULL, file = rmdfile)
}
```

**Comment by jimhester** [comment]
I think _very difficult_ is overstating it

```r
for (f in list.files(pattern = "[.]R(md)?", ignore.case = TRUE, recursive = TRUE)) {
  print(lintr::lint(f))
}
```

**Comment by mattiaciollaro** [comment]
@naupaka @jimhester Thanks again for your input!

Here is a slightly modified chunk that seems to be working on CircleCI 2, just in case someone else is trying to make it work:

```
- run:
          name: Lint R code
          command: |
            export CI=""
            Rscript \
            --slave \
            --vanilla \
            -e "files <- list.files(pattern = '[.]R', ignore.case = TRUE, recursive = TRUE); \
            lints <- ifelse(length(files) > 0, sapply(files, function(f) print(lintr::lint(f))), list()); \
            status <- as.integer(sum(sapply(lints, function(x) length(x) > 0)) > 0); \
            q('no', status = status)"
```

Note that the `export CI=""` seems to be an effective workaround to the following error

```
Error in httr::status_code(response) : object 'response' not found
Calls: print -> print.lints -> github_comment -> <Anonymous>
Execution halted
Exited with code 1
```

which otherwise pops up in my builds.

The `export CI=""` trick is taken from [this post](https://github.com/jimhester/lintr/issues/166).

**Comment by pwildenhain** [comment]
Would you be open to the creation of a `lint_dir()` function? Similar to how `styler::style_dir()`? I would be happy to tackle that one

**Comment by arekbee** [comment]
Maybe in next release  there could be a lint_dir() function?
https://github.com/jimhester/lintr/pull/360
